packages/app/src/cli/services/bulk-operations/bulk-operation-status.ts

```diff
@@ -0,0 +1,84 @@
+  GetBulkOperationByIdQuery,
+} from '../../api/graphql/bulk-operations/generated/get-bulk-operation-by-id.js'
+import {renderInfo, renderSuccess, renderError} from '@shopify/cli-kit/node/ui'
+import {outputContent, outputToken} from '@shopify/cli-kit/node/output'
+import {ensureAuthenticatedAdmin} from '@shopify/cli-kit/node/session'
+import {adminRequestDoc} from '@shopify/cli-kit/node/api/admin'
+
+const API_VERSION = '2026-01'
+
+interface GetBulkOperationStatusOptions {
+  storeFqdn: string
+  operationId: string
+}
+
+export async function getBulkOperationStatus(options: GetBulkOperationStatusOptions): Promise<void> {
+  const {storeFqdn, operationId} = options
+
+  const adminSession = await ensureAuthenticatedAdmin(storeFqdn)
+
+  const response = await adminRequestDoc<GetBulkOperationByIdQuery, {id: string}>({
+    query: GetBulkOperationById,
+    session: adminSession,
+    variables: {id: operationId},
+    version: API_VERSION,
+  })
+
+  if (response.bulkOperation) {
+    renderBulkOperationStatus(response.bulkOperation)
+  } else {
+    renderError({
+      headline: 'Bulk operation not found.',
+      body: outputContent`ID: ${outputToken.yellow(operationId)}`.value,
+    })
+  }
+}
+
+function renderBulkOperationStatus(operation: BulkOperation): void {
+  const {id, status, createdAt, completedAt, url, partialDataUrl} = operation
+  const statusDescription = formatBulkOperationStatus(operation).value
+  const timeDifference = formatTimeDifference(createdAt, completedAt)
+  const operationInfo = outputContent`ID: ${outputToken.yellow(id)}\n${timeDifference}`.value
+
+  if (status === 'COMPLETED') {
+    const downloadLink = url ? outputToken.link('Download results', url) : ''
+    renderSuccess({headline: statusDescription, body: outputContent`${operationInfo}\n${downloadLink}`.value})
+  } else if (status === 'FAILED') {
+    const downloadLink = partialDataUrl ? outputToken.link('Download partial results', partialDataUrl) : ''
+    renderError({headline: statusDescription, body: outputContent`${operationInfo}\n${downloadLink}`.value})
+  } else {
+    renderInfo({headline: statusDescription, body: operationInfo})
+  }
+}
+
+function formatTimeDifference(createdAt: unknown, completedAt?: unknown): string {
+  const now = new Date()
+
+  if (completedAt) {
+    return `Finished ${timeAgo(new Date(String(completedAt)), now)}`
+  } else {
+    return `Started ${timeAgo(new Date(String(createdAt)), now)}`
+  }
+}
+
+function timeAgo(from: Date, to: Date): string {
+  const seconds = Math.floor((to.getTime() - from.getTime()) / 1000)
+  if (seconds < 60) return `${formatTimeUnit(seconds, 'second')} ago`
+
+  const minutes = Math.floor(seconds / 60)
+  if (minutes < 60) return `${formatTimeUnit(minutes, 'minute')} ago`
+
+  const hours = Math.floor(minutes / 60)
+  if (hours < 24) return `${formatTimeUnit(hours, 'hour')} ago`
+
+  const days = Math.floor(hours / 24)
+  return `${formatTimeUnit(days, 'day')} ago`
+}
```

Comment:
This function may be useful in other places, what about moving it to `packages/cli-kit/src/public/common/string.ts`? There are similar date formatting functions there.

Reply:
Done!